azurerm/resource_arm_network_interface.go

```diff
@@ -356,7 +362,7 @@ func resourceArmNetworkInterfaceDelete(d *schema.ResourceData, meta interface{})
 		defer azureRMUnlockByName(networkSecurityGroupName, networkSecurityGroupResourceName)
 	}
 
-	configs := d.Get("ip_configuration").(*schema.Set).List()
+	configs := d.Get("ip_configuration").([]interface{})
```

Comment:
I may have tested something wrong, but this appeared to be fine when testing the migration from a resource/state created with `master`?